rb/spec/integration/selenium/webdriver/manager_spec.rb

```diff
@@ -53,103 +53,221 @@ module WebDriver
+                    reason: 'https://bugs.chromium.org/p/chromedriver/issues/detail?id=3734'} do
+          driver.get("https://saucelabs.com")
+
+          expect {
+            driver.manage.add_cookie name: 'domain',
+                                     value: 'subdomain',
+                                     domain: 'opensource.saucelabs.com'
+          }.to raise_exception(Error::InvalidCookieDomainError)
         end
 
-        it 'should add expiration with Number' do
-          driver.navigate.to url_for('xhtmlTest.html')
+        it 'should not be visible to javascript when http_only is true' do
+          driver.manage.add_cookie name: 'httponly',
+                                   value: 'true',
+                                   http_only: true
+
+          expect(driver.execute_script("return document.cookie")).to be_empty
+          expect(driver.manage.cookie_named('httponly')[:http_only]).to eq true
+        end
 
-          expected = (Date.today + 2).to_datetime
-          driver.manage.add_cookie name: 'expiration',
-                                   value: 'number',
-                                   expires: expected.to_time.to_f
+        it 'should not add secure cookie when http',
+           except: {browser: :firefox,
+                    reason: 'https://github.com/mozilla/geckodriver/issues/1840'} do
+          driver.manage.add_cookie name: 'secure',
+                                   value: 'http',
+                                   secure: true
 
-          actual = driver.manage.cookie_named('expiration')[:expires]
-          expect(actual).to be_kind_of(DateTime)
-          expect(actual).to eq(expected)
+          expect(driver.manage.all_cookies.size).to eq(0)
         end
 
-        it 'should add sameSite cookie with attribute Strict', only: {browser: %i[chrome edge firefox]} do
-          driver.navigate.to url_for('xhtmlTest.html')
-          driver.manage.add_cookie name: 'samesite', value: 'strict', same_site: 'Strict'
+        it 'should add secure cookie when https',
+           exclusive: {driver: :none,
+                       reason: "Can only be tested on https site"} do
+          driver.get 'https://www.selenium.dev'
 
-          expect(driver.manage.cookie_named('samesite')[:same_site]).to eq('Strict')
+          driver.manage.add_cookie name: 'secure',
+                                   value: 'https',
+                                   secure: true
+
+          expect(driver.manage.cookie_named('secure')[:secure]).to eq(true)
         end
 
-        it 'should add sameSite cookie with attribute Lax', only: {browser: %i[chrome edge firefox]} do
-          driver.navigate.to url_for('xhtmlTest.html')
-          driver.manage.add_cookie name: 'samesite',
-                                   value: 'lax',
-                                   same_site: 'Lax'
-          expect(driver.manage.cookie_named('samesite')[:same_site]).to eq('Lax')
+        context 'sameSite' do
```

Comment:
Question. Is this essentially showing that sameSite can be anything? Is this desirable? Or should it be some sort of set of values?

Reply:
I'm not sure what you mean. I'm using "context" just for organization in the spec. The values being tested are still just strict, lax & none

Reply:
What I mean is, what restrictions are there on the value of `sameSite` I'm not familiar with why you picked these values? Are those values the only permitted ones?

Reply:
Yes; when in doubt search the spec: https://w3c.github.io/webdriver/#dfn-cookie-same-site